Makefile

```diff
@@ -77,10 +77,27 @@ doc-build:  ## Build the docs
 	poetry run pandoc --from=markdown --to=rst --output=$(DOC_SOURCE_DIR)/tutorial.rst doc/source/tutorial.md
 	poetry run pandoc --from=markdown --to=rst --output=$(DOC_SOURCE_DIR)/user_guide.rst doc/source/user_guide.md
 	poetry run pandoc --from=markdown --to=rst --output=$(DOC_SOURCE_DIR)/changelog.rst CHANGELOG.md
-	poetry run sphinx-multiversion $(DOC_SOURCE_DIR) $(DOC_OUTPUT_DIR)
+	$(MAKE) doc-build-pydantic-v1
+	$(MAKE) doc-build-pydantic-v2
 	poetry run python doc/bin/modify_class_name.py $(DOC_OUTPUT_DIR)
 .PHONY: doc-build
 
+doc-build-pydantic-v1:
+	poetry remove pydantic autodoc-pydantic
+	poetry add 'pydantic<2' 'autodoc-pydantic<2'
+	git restore -- pyproject.toml poetry.lock # restoring straight after install makes sure that any errors during doc build will not leave them dirty
+	# We whitelist no branches, as they will be built by the pydantic-v1 target
+	poetry run sphinx-multiversion $(DOC_SOURCE_DIR) $(DOC_OUTPUT_DIR) -D 'smv_branch_whitelist=None'
+.PHONY: doc-build-pydantic-v1
```

Comment:
This is temporary ?

Reply:
Currently, we regenerate the documentation for every version. We could move to only doing new versions and not touch the old ones. In the current state, this is thus not temporary.